hoodie-client/src/main/java/com/uber/hoodie/index/bloom/HoodieBloomIndex.java

```diff
@@ -238,6 +239,11 @@ private int determineParallelism(int inputParallelism, int totalSubPartitions) {
                     return filteredFiles.iterator();
                 }).collect();
 
+        if (config.shoudCachePartitionMetadata()) {
```

Comment:
if something is just done by 1 index, then can we make the configs and property namespaces down scoped to that class alone.